src/services/user.service.ts

```diff
@@ -10,8 +17,93 @@ export class UserService {
     private readonly userRepository: Repository<User>,
   ) {}
 
+  async get(param: string): Promise<UserResponseDto[]> {
+    const users = await this.userRepository.find({
+      relations: { role: true, appointments: true },
+    });
+
+    const userResponse = users.map((user) => {
+      const userResponse = new UserResponseDto();
+      userResponse.id = user.id;
+      userResponse.name = user.name;
+      userResponse.phone = user.phone;
+      userResponse.email = user.email;
+      userResponse.role = user.role;
+      userResponse.appointments = user.appointments;
+
+      return userResponse;
+    });
+
+    if (param && userFilters[param]) {
+      return userFilters[param](users);
+    } else if (param) {
+      throw new BadRequestException(`Parámetro de filtro inválido: ${param}`);
+    }
+
+    return userResponse;
+  }
+
+  async create(createDto: UserCreateDto): Promise<User> {
+    const existingUser = await this.userRepository.findOne({
+      where: {
+        email: createDto.email,
+        phone: createDto.phone,
+        name: createDto.name,
+      },
+    });
+
+    if (existingUser) {
+      throw new Error('Usuario existente, usa diferentes credenciales para crear uno.');
+    }
+    const user = this.userRepository.create(createDto);
+    const hashedPassword = await bcryptjs.hash(createDto.password, 10);
```

Comment:
why are we using the asyncronous version of the hashing method instead of encoding the password syncronously?: `hashSync`, if you use the syncronous method you can do something like:

```suggestion
    try {
    const user = this.userRepository.create({ ...createDto, password: bcryptjs.hashSync(createDto.password, 10) });
    return await this.userRepository.save(user);
    }
```